libs/auth-nest-tools/src/lib/current-actor.decorator.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  createParamDecorator,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common'
+import { logger } from '@island.is/logging'
+
+import { User } from './user'
+import { getRequest } from './getRequest'
+
+export const getCurrentActor = (context: ExecutionContext): User => {
+  const request = getRequest(context)
+
+  const user = request.user
+
+  if (!user) {
+    logger.warn(
+      'No user authentication found. Did you forget to add IdsUserGuard?',
+    )
+    throw new UnauthorizedException()
+  }
+  return user.actor ?? user
```

Comment:
Should maybe add some of the `user` properties to the `user.actor`, so it fulfils the interface and can be used in the audit service and auth middleware.

```
return !user.actor ? user : {
  ...user.actor,
  client: user.client,
  authorization: user.authorization,
  ip: user.ip,
  userAgent: user.userAgent,
}
```

Reply:
cc @valurefugl

Reply:
Good point.